syntax = "proto3";

package expath;

message Feature {
    string type = 1;
    string value = 2;
}

message Edge {
    string source = 1;
    string destination = 2;
    string code = 3;
    int64 aggregate = 4;
    int64 outbound = 5;
    int64 inbound = 6;
    double cost = 7;

    enum ConnectionType {
        UNDEFINED = 0;
        CUSTODY = 1;
        PERSISTENT = 2;
    }

    ConnectionType type = 8;
    int64 departure = 9;
    int64 duration = 10;
    repeated Feature features = 11;
}

message Response {
    bool success = 1;
    string message = 2;
}

message FindResponse {
    Response response = 1;
    Edge edge = 2;
}

service EPServices {
    rpc AddEdge (Edge) returns (Response);
    rpc RemoveEdge (Edge) returns (Response);
    rpc SearchEdge (Edge) returns (FindResponse);
}
